tests/Unit/TestCreation.hpp

```diff
@@ -70,11 +72,15 @@ T test_creation(const std::string& construction_string) noexcept {
 /// \ingroup TestingFrameworkGroup
 /// Construct a factory object from a given string.
 ///
-/// The string must not contain the
-/// name of the option (which is the name of the struct by default if no
-/// `name()` function is present). For example, to create a struct with a
-/// `SizeT` named option the following would be used:
-/// \snippet Test_TestCreation.cpp size_t_argument
+/// The string must contain the name of the option (specifically, the class
+/// being created from options, which is the name of the struct by default if
+/// no `name()` function is present). Note that it is not the name of the base
+/// class, but the name of the derived class that must be given. For example, to
+/// create a pointer of `BaseClass` pointing to a derived class of type
```

Comment:
Here I suggest "a pointer to `BaseClass`" or "a pointer of type `BaseClass`" or maybe just "a `BaseClass*`" ... but feel free to disagree